### ISSUE TITLE
Bug Report: Tensorboard can not load all `Hyperparameters` keys, if call `writer.add_hparams` with different `hparam_dict` parameters.

### ISSUE THREAD

**Issue by FavorMylikes** [tensorflow/tensorboard#5596]
# Test code

```python
from torch.utils.tensorboard import SummaryWriter

from conf import WRITER_PATH

writer = SummaryWriter(str(WRITER_PATH / "test1"))
with writer:
    writer.add_hparams(hparam_dict={"height": 0}, metric_dict={"iou": 0.5})
    writer.flush()

writer = SummaryWriter(str(WRITER_PATH / "test2"))
with writer:
    writer.add_hparams(hparam_dict={"weight": 0}, metric_dict={"iou": 0.3})
    writer.flush()
```

# response

- We can see all `hparams` from response with `height` and `weight`
<details>
  <summary>Json response from /data/plugin/hparams/session_groups</summary>
```json
{
  "sessionGroups": [
    {
      "name": "test1/1646371009.676503",
      "hparams": {
        "height": 0.0
      },
      "metricValues": [
        {
          "name": {
            "tag": "iou",
            "group": ""
          },
          "value": 0.5,
          "wallTimeSecs": 1646371009.6773322,
          "trainingStep": 0
        }
      ],
      "sessions": [
        {
          "name": "test1/1646371009.676503",
          "status": "STATUS_SUCCESS",
          "metricValues": [
            {
              "name": {
                "tag": "iou",
                "group": ""
              },
              "value": 0.5,
              "wallTimeSecs": 1646371009.6773322,
              "trainingStep": 0
            }
          ],
          "startTimeSecs": 0.0,
          "endTimeSecs": 0.0,
          "modelUri": "",
          "monitorUrl": ""
        }
      ],
      "monitorUrl": ""
    },
    {
      "name": "test2/1646371009.6787846",
      "hparams": {
        "weight": 0.0
      },
      "metricValues": [
        {
          "name": {
            "tag": "iou",
            "group": ""
          },
          "value": 0.30000001192092896,
          "wallTimeSecs": 1646371009.679453,
          "trainingStep": 0
        }
      ],
      "sessions": [
        {
          "name": "test2/1646371009.6787846",
          "status": "STATUS_SUCCESS",
          "metricValues": [
            {
              "name": {
                "tag": "iou",
                "group": ""
              },
              "value": 0.30000001192092896,
              "wallTimeSecs": 1646371009.679453,
              "trainingStep": 0
            }
          ],
          "startTimeSecs": 0.0,
          "endTimeSecs": 0.0,
          "modelUri": "",
          "monitorUrl": ""
        }
      ],
      "monitorUrl": ""
    }
  ],
  "totalSize": 2
}
```
</details>

# Page

Here the weight **ONLY**

![image](https://user-images.githubusercontent.com/16064851/156707191-baf8a848-0b63-4443-a7fa-d9543e05e8bb.png)

# Version

```console
> tensorboard --version                                                                                                                                           (py38) 
TensorFlow installation not found - running with reduced feature set.
2.8.0
```

**Comment by yatbear** [comment]
Hi @FavorMylikes,

Can you please share the full command you used to run the TensorBoard? At first glance it looks like TensorBoard only loaded the second logdir which is `str(WRITER_PATH / "test2")`.

**Comment by FavorMylikes** [comment]
@yatbear 

```console
tensorboard --logdir=runs --host=::0 --port 6006
```

Here runs is the Path to `WRITER_PATH`

And `test1/` at the first row and `Trial ID` column in hparams table view which we can see.

**Comment by yatbear** [comment]
Update:
The keys are missing due to the following reason: on TensorBoard side, hparams across runs do get merged if you **don't** explicitly specify an `experiment`, but the PyTorch API forces logging an `experiment` explicitly [here](https://github.com/pytorch/pytorch/blob/7ed73b280352c20257300f56e1c0b6a21a616734/torch/utils/tensorboard/summary.py#L56). Since you didn't specify an experiment in the code, using PyTorch API to add_hparams is equivalent to having 2 different experiments for each set of your hparams (`weight` and `height`), and only one set was loaded since the HParams dashboard doesn't support multiple experiments yet.

**Comment by n-gao** [comment]
Is there any update on this? During development, it may happen that new parameters are added or dropped. Tensorboard only ever lists the hyperparameters of the very first run.

**Comment by yatbear** [comment]
Hi @n-gao,

Did you try the suggestion in the comment above: https://github.com/tensorflow/tensorboard/issues/5596#issuecomment-1061021673? If it didn't help, can you please provide some exemplary data and steps to reproduce your issue? Thanks!

**Comment by Kerybas** [comment]
Hello @yatbear, I ran some tests here, if you want to have a look:
https://github.com/pytorch/pytorch/issues/39250#issuecomment-1304883545  
I think I addressed your point about the experiment name by fixing `log_dir="runs/ABC"`, didn't I?

**Comment by bmd3k** [comment]
Hi @Kerybas, I think @yatbear's advice for a workaround was to try to log the same hparams for all the experiments. In your example this would look something like `w.add_hparams({"key_A": 1, "key_B": 2, "key_C": 3}, {"loss": 4})`.
